src/shell.js

```diff
@@ -146,7 +146,7 @@ var ENVIRONMENT_IS_WASM_WORKER = Module['$ww'];
 #if SHARED_MEMORY && !MODULARIZE
 // In MODULARIZE mode _scriptDir needs to be captured already at the very top of the page immediately when the page is parsed, so it is generated there
 // before the page load. In non-MODULARIZE modes generate it here.
-var _scriptDir = (typeof document != 'undefined' && document.currentScript) ? document.currentScript.src : undefined;
+var _scriptDir = (typeof document != 'undefined') ? document.currentScript?.src : undefined;
```

Comment:
Even shorter:
```suggestion
var _scriptDir = document?.currentScript?.src;
```

Reply:
I don't think `foo?` work for global sadly.  i.e. you can't do `foo?.method()` but you can do `globalThis.foo?.method()`.

Maybe if we knew we have univeral access to globalThis we should use that here, but I'm not sure we can assume that yet

Reply:
Huh. I thought it would work, but didn't test sorry.